### PR TITLE
Fix configure when lint tools are missing. Closes #1820

### LINKED ISSUE #1820: CPPLint build fail
I'm getting this build failre with the latest master branch:

21:53:51 -------------------------------------------------------
21:53:51 OLA Version 0.10.8
21:53:51 
21:53:51 Prefix: '/usr/local'
21:53:51 Compiler: 'g++ -g -O2 -std=gnu++11 -pthread '
21:53:51 Linker: '/usr/bin/ld -m elf_x86_64   -ldl '
21:53:51 Python: /usr/bin/python
21:53:51 
21:53:51 Python API: yes
21:53:51 Java API: no
21:53:51 Enable HTTP Server: yes
21:53:51 RDM Responder Tests: yes
21:53:51 Ja Rule: no
21:53:51 Enabled Plugins: artnet dummy e131 espnet ftdidmx gpio karate kinet milinst nanoleaf opendmx openpixelcontrol osc pathport renard sandnet shownet spi spidmx stageprofi uartdmx usbdmx usbpro
21:53:51 UUCP Lock Directory: /var/lock
21:53:51 
21:53:51 Now type 'make [<target>]'
21:53:51   where the optional <target> is:
21:53:51     all          - build everything
21:53:51     check        - run the tests
21:53:51     doxygen-doc  - generate the HTML documentation
21:53:51     lint         - run the linters
21:53:51 -------------------------------------------------------
21:53:51 [Open_Lighting_Architecture_-_Linux] $ /bin/bash -xe /tmp/jenkins6349608851020370006.sh
21:53:51 + make all
21:53:51 Makefile:20124: *** cpplint not found. Install forked cpplint (e.g. via pip for the latest version) and re-run configure.. Stop.

Git hash is 95b8cbecd9bd18a2b4436cd799c0cc80c4795b95
build commands from fresh clone are:
autoreconf -i
./configure --enable-rdm-tests --disable-root-check --disable-dependency-tracking
make all

Looking at the generated makefile I should be able to build without cpplint, but this does not seem possible at the moment?

### PR DESCRIPTION
**Pull request by peternewman**
Closes #1820

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -261,5 +261,5 @@ cpplint: Makefile.am
 if FOUND_CPPLINT
 	cpplint --filter=$(CPP_LINT_FILTER) $(CPP_LINT_FILES)
 else
-  $(error cpplint not found. Install forked cpplint (e.g. via pip for the latest version) and re-run configure.)
+	$(error cpplint not found. Install the forked cpplint (e.g. via pip for the latest version) and re-run configure.)
```
Comment (DaAwesomeP) [inline Makefile.am:264]: Wow I would never have found that!

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -23,8 +23,8 @@ jobs:
           apt-get -y install pkg-config libtool autoconf \
             automake g++ bison flex make bash-completion dh-autoreconf \
             debhelper devscripts wget python3-pip
-      - name: Install Python lint tools
-        run: python3 -m pip install --no-input cpplint flake8
+      #- name: Install Python lint tools
+      #  run: python3 -m pip install --no-input cpplint flake8
```
Comment (DaAwesomeP) [inline .github/workflows/lint.yaml:27]: Should be able to remove this for all unrelated tasks now (except for of course flake8 in flake8 and cpplint in cpplint).

Comment (DaAwesomeP) [reply .github/workflows/lint.yaml:27]: Although I suppose you do need them when building or you will have to reconfigure.

Comment (peternewman) [reply .github/workflows/lint.yaml]: Yeah exactly that hence why it was failing just now when the build for lint passed. I only removed them to confirm configure succeeded with them not present and didn't crash.

File: Makefile.am (hunk 1 above)
Comment (peternewman) [reply Makefile.am:264]: TBH configure on GitHub actions which I guess is newer gave an even more ominous message (or it's just the fact its a clean checkout each time).

It was pretty blatantly obviously different when I went to edit the file on GitHub so I gave it a go and the rest is history.

**Review by DaAwesomeP — CHANGES_REQUESTED** [review]
You should now be able to remove flake8 from here:

https://github.com/OpenLightingProject/ola/blob/7b4a27361db2cd2ec46ff21d84cf82f4a589ee3c//.github/workflows/lint.yaml#L143

And cpplint from here:

https://github.com/OpenLightingProject/ola/blob/7b4a27361db2cd2ec46ff21d84cf82f4a589ee3c//.github/workflows/lint.yaml#L185

**Comment by peternewman** [comment]
> You should now be able to remove flake8 from here:

Managed to be even more drastic!

**Review by DaAwesomeP — APPROVED** [review]
Amazing!

**Comment by rodb70** [comment]
@peternewman 
I get this error message at the configure stage:
checking for Cppunit - version >= 1.9.6... no
configure: error: Missing cppunit, please install >= 1.9.6

libcppunit-dev is already the newest version (1.15.1-2build1).

**Comment by peternewman** [comment]
> @peternewman I get this error message at the configure stage:
> checking for Cppunit - version >= 1.9.6... no configure: error: Missing cppunit, please install >= 1.9.6

So its getting further/past that cpplint/flake8 stage now at least @rodb70 ?

> libcppunit-dev is already the newest version (1.15.1-2build1).

Strange, that's the exact version we're using in our lint where it works fine. Please could you share your config.log with us?

**Comment by rodb70** [comment]
@peternewman, my original failure was at the beginning on make.  This configure issue was earlier in the buuild process at the configure stage.

**Comment by peternewman** [comment]
Hmm, the only change we've made to configure.ac since December 2021 was to look for cpplint rather than cpplint.py, which obviously in itself won't have broken cppunit detection, but may have caused configure to re-run/regenerate, potentially flagging an existing issue you had.

As mentioned we'd need to see your config.log file to do any further useful diagnostics given it's working in various other places.

I'll merge this PR as it improves things for the rest of us, but please follow up here or in the issue about cppunit just in case we've somehow inadvertently broken things.

**Comment by rodb70** [comment]
@peternewman After merge build on master ok.  Thank you.

**Pull request merged**